src/banded/dot.jl

```diff
@@ -0,0 +1,43 @@
+function _dot(x::AbstractVector, A::BandedMatrix, y::AbstractVector)
+    require_one_based_indexing(x, y)
+    (axes(x)..., axes(y)...) == axes(A) || throw(DimensionMismatch())
+    l,u = bandwidths(A)
+    M = size(A,1)
+    T = typeof(dot(first(x), first(A), first(y)))
+    s = zero(T)
+    i₁ = first(eachindex(x))
+    x₁ = first(x)
+    @inbounds for j in eachindex(y)
```

Comment:
Note `A[i,j]` is slow so this will be a _lot_ faster if written in terms of `A.data`

Reply:
I feared as much 😁

Reply:
Alternatively one could do some work on ` @inbands ` macro

Reply:
Actually: leave it as is but only restrict to `AbstractBandedMatrix`: then you have the excuse that you don't know that `A.data` exists 😀

Reply:
Sure, but it will not be magically faster this way, right? :ton